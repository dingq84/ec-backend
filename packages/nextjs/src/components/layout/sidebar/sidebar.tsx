import Link from 'next/link'
import Image from 'next/image'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faChevronRight } from '@fortawesome/free-solid-svg-icons'
import tw, { css } from 'twin.macro'

// components
import Outer from '@/components/layout/sidebar/outer'
import Collapse from '@/components/shared/collapse'

// states
import { useAppDispatch, useAppSelector } from '@/states/hooks'
import { toggleSidebar } from '@/states/settings'

const backgroundGradient = css`
  & {
    background: linear-gradient(183.74deg, #191b2c -30.18%, #161054 59.34%, #393091 110.71%);
  }
`

const Sidebar = () => {
  const dispatch = useAppDispatch()
  const sidebarIsExtend = useAppSelector(state => state.settings.sidebarIsExtend)
  const sidebarMenu = useAppSelector(state => state.me.menu)

  const handleClick = (): void => {
    dispatch(toggleSidebar())
  }

  return (
    <aside css={[backgroundGradient]} tw="py-5 flex flex-col flex-shrink-0 px-3">
      <button
        className="flex-center btn-outline"
        onClick={handleClick}
        tw="absolute transition-all duration-300 left-52 top-16 rounded-full text-gray-3 bg-white p-0 w-6 h-6 transform -translate-x-1/2 -translate-y-1/2 shadow-md border border-solid border-gray-1 hover:(text-blue-1 border-purple-1)"
        css={[
          sidebarIsExtend === false && tw`left-20 -translate-x-full`,
          sidebarMenu.length === 0 && tw`translate-x-1/2`
        ]}
        role="presentation"
      >
        <FontAwesomeIcon
          icon={faChevronRight}
          tw="color[inherit] text-xs transition-transform duration-300"
          css={[sidebarIsExtend && tw`transform rotate-180`]}
        />
      </button>

      <Link href="/">
        <a tw="inline-flex justify-start mb-14 pl-3">
          <Image src="/images/logo.svg" alt="logo" width={24} height={20} />
          <Collapse inProps={sidebarIsExtend} tw="ml-2" orientation="horizontal">
            <Image src="/images/logoWord.svg" alt="vivy admin website" width={64} height={18} />
          </Collapse>
        </a>
      </Link>

      <div className="scroll-y">
        <ul
          tw="flex-grow"
          css={[sidebarIsExtend && tw`mr-3`, sidebarMenu.length === 0 && tw`w-48`]}
        >
          {sidebarMenu.map(menu => (
            <Outer menu={menu} key={menu.id} />
          ))}
        </ul>
      </div>
    </aside>
  )
}

export default Sidebar
