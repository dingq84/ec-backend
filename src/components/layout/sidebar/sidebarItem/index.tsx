export { default } from './sidebarItem'
export type { SidebarItemProps } from './sidebarItem'
