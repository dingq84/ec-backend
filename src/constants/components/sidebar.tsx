/**
 * @author Ding.Chen 2021-06-18
 * 後台完整 sidebar menu，根據 api 提供的權限做過濾
 */
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faUserCog } from '@fortawesome/free-solid-svg-icons'

const SIDEBAR_MENU = [
  {
    prefix: <FontAwesomeIcon icon={faUserCog} />,
    name: '訂單管理',
    href: '/'
  },
  {
    prefix: <FontAwesomeIcon icon={faUserCog} />,
    name: '會員管理',
    children: [
      {
        prefix: <FontAwesomeIcon icon={faUserCog} />,
        name: '會員列表與詳細頁',
        href: '/account/info',
        children: [
          {
            prefix: <FontAwesomeIcon icon={faUserCog} />,
            name: '會員列表與詳細頁',
            href: '/account/info/a'
          },
          {
            prefix: <FontAwesomeIcon icon={faUserCog} />,
            name: '匯出會員資訊',
            href: '/account/info/b'
          }
        ]
      },
      {
        prefix: <FontAwesomeIcon icon={faUserCog} />,
        name: '匯出會員資訊',
        href: '/account/export'
      }
    ]
  },
  {
    prefix: <FontAwesomeIcon icon={faUserCog} />,
    name: '商品管理',
    href: '/product'
  },
  {
    prefix: <FontAwesomeIcon icon={faUserCog} />,
    name: '促銷管理',
    href: '/promote'
  }
]

export default SIDEBAR_MENU
