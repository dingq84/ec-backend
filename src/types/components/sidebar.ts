type ConstantsSidebarMenuType = {
  prefix: any
  name: string
  href?: string
  children?: ConstantsSidebarMenuType[]
}

type SidebarMenuType = Omit<ConstantsSidebarMenuType, 'children'> & {
  isOpen?: boolean
  isActive?: boolean
  suffix?: any
  key?: string // 由 index 組成
  children?: SidebarMenuType[]
}

type SidebarItemProps = {
  sidebarItems: SidebarMenuType[]
  toggleSidebarOpen: (key: string, open?: boolean) => void
  forwardTo: (href: string) => void
}

export type { ConstantsSidebarMenuType, SidebarMenuType, SidebarItemProps }
