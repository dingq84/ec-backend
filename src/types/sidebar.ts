/**
 * @author Dean Chen 2021-04-27
 * 1. BASIC_SIDEBAR_MENU_TYPE 提供給使用者定義 sidebar 的設定型態，
 * 2. SIDEBAR_MENU_TYPE 會針對使用者設定的 sidebar 新增多個屬性以便後續使用
 */

import { IconDefinition } from '@fortawesome/fontawesome-svg-core'

type BASIC_SIDEBAR_MENU_TYPE = {
  icon?: IconDefinition
  name: string // the name of menu
  children?: Array<BASIC_SIDEBAR_MENU_TYPE> // menu's children
  href?: string // the href of menu item
}

type SIDEBAR_MENU_TYPE = BASIC_SIDEBAR_MENU_TYPE & {
  key: string // the key of menu
  isActive: boolean // whether the menu is active or not
  isOpen: boolean // whether the menu is open or not
  level: number // the level of the menu. the menu item is more low, it's level is more high
  children?: Array<SIDEBAR_MENU_TYPE> // menu's children
}

export type { BASIC_SIDEBAR_MENU_TYPE, SIDEBAR_MENU_TYPE }
