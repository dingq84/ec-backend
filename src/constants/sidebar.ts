/**
 * @author Dean Chen 2021-04-12
 * BASIC_SIDEBAR_MENU 提供使用者自定義 sidebar，至於 sidebar 畫面的邏輯，會透過 utils/sidebar 去做處理
 * TODO: 根據未來專案需求，可能會有權限設定
 */

import { faTachometerAlt } from '@fortawesome/free-solid-svg-icons'

// types
import { BASIC_SIDEBAR_MENU_TYPE } from '@/types/sidebar'

const BASIC_SIDEBAR_MENU: Array<BASIC_SIDEBAR_MENU_TYPE> = [
  {
    icon: faTachometerAlt,
    name: 'Dashboard',
    children: [
      {
        icon: faTachometerAlt,
        name: 'Dashboard v1',
        children: [
          {
            icon: faTachometerAlt,
            name: 'Dashboard v1 v1',
            children: [
              {
                icon: faTachometerAlt,
                name: 'Dashboard v1 v1 v1',
                href: '/test'
              },
              {
                icon: faTachometerAlt,
                name: 'Dashboard v1 v1 v2',
                href: '/'
              }
            ]
          },
          {
            icon: faTachometerAlt,
            name: 'Dashboard v1 v2',
            href: '/'
          }
        ]
      },
      {
        icon: faTachometerAlt,
        name: 'Dashboard v2',
        href: '/'
      }
    ]
  },
  {
    icon: faTachometerAlt,
    name: 'About'
  }
]

export { BASIC_SIDEBAR_MENU }
