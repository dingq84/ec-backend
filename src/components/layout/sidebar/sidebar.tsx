/**
 * @author Dean Chen 2021-04-27
 * Sidebar 完成基本的 menu tree 顯示，並加上桌機版時，可縮小 sidebar 變成 hover 效果
 * TODO: 需新增判斷 route path，找出目前的路徑
 */

import { useRouter } from 'next/router'
import { SyntheticEvent, useState, useRef, useCallback } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons'
import tw from 'twin.macro'

// components
import Collapse from '@/components/common/collapse'
import Popover from '@/components/common/popover'

// constants
import { BASIC_SIDEBAR_MENU } from '@/constants/sidebar'

// hooks
import useEnhancedEffect from '@/hooks/useEnhancedEffect'
import useIsMobile from '@/hooks/useIsMobile'

// states
import { useAppSelector } from '@/states/global/hooks'

// types
import { SIDEBAR_MENU_TYPE } from '@/types/sidebar'

// utils
import addProperties from '@/utils/sidebar/addProperties'
import getIsOpen from '@/utils/sidebar/getIsOpen'
import getPreviousKey from '@/utils/sidebar/getPreviousKey'

type SidebarItemProps = {
  item: SIDEBAR_MENU_TYPE
  toggleActiveMenuItem: Function
  isFloat: boolean
}

const SidebarItem = (props: SidebarItemProps) => {
  const anchorEl = useRef<HTMLDivElement>(null)
  const router = useRouter()
  const { item, toggleActiveMenuItem, isFloat } = props
  const { isOpen, key, name, icon, isActive, href, children, level } = item
  const isFirstLevel = level === 1
  const isFirstLevelAndFloat = isFloat && isFirstLevel
  const isFirstLevelAndOpen = isOpen && isFirstLevel
  const hasChildren = Boolean(children)

  let Transition: any = Collapse
  let transitionProps: Object = { inProps: isOpen }
  if (isFirstLevelAndFloat) {
    Transition = Popover
    transitionProps = {
      ...transitionProps,
      anchorEl: anchorEl.current,
      anchorOrigin: {
        horizontal: 'left',
        vertical: 'top'
      },
      hiddenBackdrop: true,
      horizontalSpace: 48,
      paperProps: {
        css: [tw`p-0 bg-dark-blue-1 rounded-l-none flex-col w-60`]
      }
    }
  }

  const getLiProps = () => {
    if (isFirstLevelAndFloat) {
      return {
        onMouseEnter(event: SyntheticEvent) {
          toggleActiveMenuItem(event, key)
        },
        onMouseLeave(event: SyntheticEvent) {
          toggleActiveMenuItem(event, '')
        }
      }
    }

    return {
      onClick(event: SyntheticEvent) {
        toggleActiveMenuItem(event, key)

        if (href) {
          router.push(href)
        }
      }
    }
  }

  return (
    <li
      {...getLiProps()}
      css={[
        tw`text-light-blue-4 text-sm cursor-pointer`,
        isFirstLevel === false && tw`bg-dark-blue-5 text-light-blue-1`
      ]}
    >
      <div
        ref={anchorEl}
        className={`flex items-center justify-start py-3 pr-4 transition-all pl-${key.length + 2}`}
        css={[
          href && isActive && tw`text-white`,
          isFirstLevelAndOpen &&
            tw`bg-dark-blue-4 border-l-4 border-light-blue-2 border-solid text-white`
        ]}
      >
        {icon ? (
          <FontAwesomeIcon
            icon={icon}
            className={`${isFirstLevelAndFloat ? 'text-md' : 'text-sm mr-2'}`}
          />
        ) : null}
        {isFirstLevelAndFloat ? null : (
          <>
            <span tw="select-none text-sm flex-grow">{name}</span>
            {hasChildren ? (
              <FontAwesomeIcon
                icon={faChevronLeft}
                tw="text-xs transition-transform"
                className={`transform ${isOpen ? '-rotate-90' : 'rotate-0'}`}
              />
            ) : null}
          </>
        )}
      </div>

      <Transition {...transitionProps}>
        {isFirstLevelAndFloat ? (
          <span
            css={[tw`text-light-blue-1 py-3 pr-4 text-sm w-full`, isOpen && tw`text-white`]}
            className={`flex items-center justify-start py-3 pr-4 transition-all pl-5`}
          >
            {name}
          </span>
        ) : null}
        {hasChildren ? (
          <ul tw="w-full">
            {children?.map((kid: SIDEBAR_MENU_TYPE) => (
              <SidebarItem
                key={kid.key}
                isFloat={isFloat}
                toggleActiveMenuItem={toggleActiveMenuItem}
                item={kid}
              />
            ))}
          </ul>
        ) : null}
      </Transition>
    </li>
  )
}

const Sidebar: React.FC = () => {
  const isMobile = useIsMobile()
  const [activeMenuKey, setActiveMenuKey] = useState('')
  const [menuList, setMenuList] = useState<SIDEBAR_MENU_TYPE[]>([])
  const sidebarIsExtend = useAppSelector(state => state.settings.sidebarIsExtend)
  const isDesktopAndCollapsed = isMobile === false && sidebarIsExtend === false
  // TODO: 這邊有個問題，當桌機縮成 48px，螢幕尺寸變成手機時，會出現手機的縮放仍是 48px
  const collapsedSize = isMobile ? '0px' : '48px'

  // 當 activeMenuKey 更新，重新整理 menuList 的 isOpen、isActive 等 key
  useEnhancedEffect(() => {
    setMenuList(addProperties(BASIC_SIDEBAR_MENU, activeMenuKey))
  }, [activeMenuKey])

  // reset active key
  useEnhancedEffect(() => {
    setActiveMenuKey('')
  }, [sidebarIsExtend])

  const toggleActiveMenuItem = useCallback(
    (event: SyntheticEvent, itemKey: string): void => {
      event.stopPropagation()

      let newActiveMenuKey: string = itemKey
      if (getIsOpen(itemKey, activeMenuKey)) {
        newActiveMenuKey = getPreviousKey(itemKey)
      }

      setActiveMenuKey(newActiveMenuKey)
    },
    [activeMenuKey]
  )

  return (
    <Collapse
      inProps={sidebarIsExtend}
      orientation="horizontal"
      tw="flex-shrink-0"
      collapsedSize={collapsedSize}
      timeout={100}
    >
      <aside tw="bg-dark-blue-1 w-60 min-height[calc(100vh - 6rem)] md:(min-height[calc(100vh - 3rem)])">
        <ul>
          {menuList.map(item => (
            <SidebarItem
              key={item.key}
              item={item}
              isFloat={isDesktopAndCollapsed}
              toggleActiveMenuItem={toggleActiveMenuItem}
            />
          ))}
        </ul>
      </aside>
    </Collapse>
  )
}

export default Sidebar
