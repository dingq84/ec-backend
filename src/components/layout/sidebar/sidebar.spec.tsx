import { render } from '@testing-library/react'

// components
import Sidebar from '.'

// constants
import * as constants from '@/constants/sidebar'

jest.mock('@/states/global/hooks')

describe('test sidebar', () => {
  it('should render correctly', () => {
    // @ts-ignore
    // https://stackoverflow.com/questions/42977961/how-to-mock-an-exported-const-in-jest
    constants.BASIC_SIDEBAR_MENU = [
      {
        name: 'Dashboard',
        children: [
          {
            name: 'Dashboard v1',
            children: [
              {
                name: 'Dashboard v1 v1',
                children: [
                  {
                    name: 'Dashboard v1 v1 v1',
                    href: '/test'
                  }
                ]
              },
              {
                name: 'Dashboard v1 v2',
                href: '/'
              }
            ]
          },
          {
            name: 'Dashboard v2',
            href: '/'
          }
        ]
      },
      {
        name: 'About'
      }
    ]
    const tree = render(<Sidebar />)
    expect(tree).toMatchSnapshot()
  })
})
