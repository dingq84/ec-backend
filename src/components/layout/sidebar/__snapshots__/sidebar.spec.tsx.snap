// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing sidebar should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="collapse___StyledDiv-deqgwy-0 bJjHov sidebar___StyledCollapse-sc-15m1zdu-6 jIpnwq"
        role="root"
        style="min-width: 48px; width: 0px;"
      >
        <div
          role="wrapper"
        >
          <aside
            class="sidebar___StyledAside-sc-15m1zdu-7 hzJcaN"
          >
            <ul>
              <li
                class="sidebar___StyledLi-sc-15m1zdu-0 eEWGUg"
              >
                <div
                  class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-3"
                >
                  <span
                    class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                  >
                    Dashboard
                  </span>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-chevron-left fa-w-10 sidebar___StyledFontAwesomeIcon-sc-15m1zdu-3 gTgPKq transform rotate-0"
                    data-icon="chevron-left"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="collapse___StyledDiv-deqgwy-0 dISutg"
                  role="root"
                  style="min-height: 0px; max-height: 0px;"
                >
                  <div
                    role="wrapper"
                  >
                    <ul
                      class="sidebar___StyledUl-sc-15m1zdu-5 gMbmau"
                    >
                      <li
                        class="sidebar___StyledLi-sc-15m1zdu-0 cmTzXe"
                      >
                        <div
                          class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-5"
                        >
                          <span
                            class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                          >
                            Dashboard v1
                          </span>
                          <svg
                            aria-hidden="true"
                            class="svg-inline--fa fa-chevron-left fa-w-10 sidebar___StyledFontAwesomeIcon-sc-15m1zdu-3 gTgPKq transform rotate-0"
                            data-icon="chevron-left"
                            data-prefix="fas"
                            focusable="false"
                            role="img"
                            viewBox="0 0 320 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                              fill="currentColor"
                            />
                          </svg>
                        </div>
                        <div
                          class="collapse___StyledDiv-deqgwy-0 dISutg"
                          role="root"
                          style="min-height: 0px; max-height: 0px;"
                        >
                          <div
                            role="wrapper"
                          >
                            <ul
                              class="sidebar___StyledUl-sc-15m1zdu-5 gMbmau"
                            >
                              <li
                                class="sidebar___StyledLi-sc-15m1zdu-0 cmTzXe"
                              >
                                <div
                                  class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-7"
                                >
                                  <span
                                    class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                                  >
                                    Dashboard v1 v1
                                  </span>
                                  <svg
                                    aria-hidden="true"
                                    class="svg-inline--fa fa-chevron-left fa-w-10 sidebar___StyledFontAwesomeIcon-sc-15m1zdu-3 gTgPKq transform rotate-0"
                                    data-icon="chevron-left"
                                    data-prefix="fas"
                                    focusable="false"
                                    role="img"
                                    viewBox="0 0 320 512"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                                      fill="currentColor"
                                    />
                                  </svg>
                                </div>
                                <div
                                  class="collapse___StyledDiv-deqgwy-0 dISutg"
                                  role="root"
                                  style="min-height: 0px; max-height: 0px;"
                                >
                                  <div
                                    role="wrapper"
                                  >
                                    <ul
                                      class="sidebar___StyledUl-sc-15m1zdu-5 gMbmau"
                                    >
                                      <li
                                        class="sidebar___StyledLi-sc-15m1zdu-0 cmTzXe"
                                      >
                                        <div
                                          class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-9"
                                        >
                                          <span
                                            class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                                          >
                                            Dashboard v1 v1 v1
                                          </span>
                                        </div>
                                        <div
                                          class="collapse___StyledDiv-deqgwy-0 dISutg"
                                          role="root"
                                          style="min-height: 0px; max-height: 0px;"
                                        >
                                          <div
                                            role="wrapper"
                                          />
                                        </div>
                                      </li>
                                    </ul>
                                  </div>
                                </div>
                              </li>
                              <li
                                class="sidebar___StyledLi-sc-15m1zdu-0 cmTzXe"
                              >
                                <div
                                  class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-7"
                                >
                                  <span
                                    class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                                  >
                                    Dashboard v1 v2
                                  </span>
                                </div>
                                <div
                                  class="collapse___StyledDiv-deqgwy-0 dISutg"
                                  role="root"
                                  style="min-height: 0px; max-height: 0px;"
                                >
                                  <div
                                    role="wrapper"
                                  />
                                </div>
                              </li>
                            </ul>
                          </div>
                        </div>
                      </li>
                      <li
                        class="sidebar___StyledLi-sc-15m1zdu-0 cmTzXe"
                      >
                        <div
                          class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-5"
                        >
                          <span
                            class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                          >
                            Dashboard v2
                          </span>
                        </div>
                        <div
                          class="collapse___StyledDiv-deqgwy-0 dISutg"
                          role="root"
                          style="min-height: 0px; max-height: 0px;"
                        >
                          <div
                            role="wrapper"
                          />
                        </div>
                      </li>
                    </ul>
                  </div>
                </div>
              </li>
              <li
                class="sidebar___StyledLi-sc-15m1zdu-0 eEWGUg"
              >
                <div
                  class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-3"
                >
                  <span
                    class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                  >
                    About
                  </span>
                </div>
                <div
                  class="collapse___StyledDiv-deqgwy-0 dISutg"
                  role="root"
                  style="min-height: 0px; max-height: 0px;"
                >
                  <div
                    role="wrapper"
                  />
                </div>
              </li>
            </ul>
          </aside>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="collapse___StyledDiv-deqgwy-0 bJjHov sidebar___StyledCollapse-sc-15m1zdu-6 jIpnwq"
      role="root"
      style="min-width: 48px; width: 0px;"
    >
      <div
        role="wrapper"
      >
        <aside
          class="sidebar___StyledAside-sc-15m1zdu-7 hzJcaN"
        >
          <ul>
            <li
              class="sidebar___StyledLi-sc-15m1zdu-0 eEWGUg"
            >
              <div
                class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-3"
              >
                <span
                  class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                >
                  Dashboard
                </span>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-chevron-left fa-w-10 sidebar___StyledFontAwesomeIcon-sc-15m1zdu-3 gTgPKq transform rotate-0"
                  data-icon="chevron-left"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 320 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <div
                class="collapse___StyledDiv-deqgwy-0 dISutg"
                role="root"
                style="min-height: 0px; max-height: 0px;"
              >
                <div
                  role="wrapper"
                >
                  <ul
                    class="sidebar___StyledUl-sc-15m1zdu-5 gMbmau"
                  >
                    <li
                      class="sidebar___StyledLi-sc-15m1zdu-0 cmTzXe"
                    >
                      <div
                        class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-5"
                      >
                        <span
                          class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                        >
                          Dashboard v1
                        </span>
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-chevron-left fa-w-10 sidebar___StyledFontAwesomeIcon-sc-15m1zdu-3 gTgPKq transform rotate-0"
                          data-icon="chevron-left"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 320 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                            fill="currentColor"
                          />
                        </svg>
                      </div>
                      <div
                        class="collapse___StyledDiv-deqgwy-0 dISutg"
                        role="root"
                        style="min-height: 0px; max-height: 0px;"
                      >
                        <div
                          role="wrapper"
                        >
                          <ul
                            class="sidebar___StyledUl-sc-15m1zdu-5 gMbmau"
                          >
                            <li
                              class="sidebar___StyledLi-sc-15m1zdu-0 cmTzXe"
                            >
                              <div
                                class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-7"
                              >
                                <span
                                  class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                                >
                                  Dashboard v1 v1
                                </span>
                                <svg
                                  aria-hidden="true"
                                  class="svg-inline--fa fa-chevron-left fa-w-10 sidebar___StyledFontAwesomeIcon-sc-15m1zdu-3 gTgPKq transform rotate-0"
                                  data-icon="chevron-left"
                                  data-prefix="fas"
                                  focusable="false"
                                  role="img"
                                  viewBox="0 0 320 512"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                                    fill="currentColor"
                                  />
                                </svg>
                              </div>
                              <div
                                class="collapse___StyledDiv-deqgwy-0 dISutg"
                                role="root"
                                style="min-height: 0px; max-height: 0px;"
                              >
                                <div
                                  role="wrapper"
                                >
                                  <ul
                                    class="sidebar___StyledUl-sc-15m1zdu-5 gMbmau"
                                  >
                                    <li
                                      class="sidebar___StyledLi-sc-15m1zdu-0 cmTzXe"
                                    >
                                      <div
                                        class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-9"
                                      >
                                        <span
                                          class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                                        >
                                          Dashboard v1 v1 v1
                                        </span>
                                      </div>
                                      <div
                                        class="collapse___StyledDiv-deqgwy-0 dISutg"
                                        role="root"
                                        style="min-height: 0px; max-height: 0px;"
                                      >
                                        <div
                                          role="wrapper"
                                        />
                                      </div>
                                    </li>
                                  </ul>
                                </div>
                              </div>
                            </li>
                            <li
                              class="sidebar___StyledLi-sc-15m1zdu-0 cmTzXe"
                            >
                              <div
                                class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-7"
                              >
                                <span
                                  class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                                >
                                  Dashboard v1 v2
                                </span>
                              </div>
                              <div
                                class="collapse___StyledDiv-deqgwy-0 dISutg"
                                role="root"
                                style="min-height: 0px; max-height: 0px;"
                              >
                                <div
                                  role="wrapper"
                                />
                              </div>
                            </li>
                          </ul>
                        </div>
                      </div>
                    </li>
                    <li
                      class="sidebar___StyledLi-sc-15m1zdu-0 cmTzXe"
                    >
                      <div
                        class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-5"
                      >
                        <span
                          class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                        >
                          Dashboard v2
                        </span>
                      </div>
                      <div
                        class="collapse___StyledDiv-deqgwy-0 dISutg"
                        role="root"
                        style="min-height: 0px; max-height: 0px;"
                      >
                        <div
                          role="wrapper"
                        />
                      </div>
                    </li>
                  </ul>
                </div>
              </div>
            </li>
            <li
              class="sidebar___StyledLi-sc-15m1zdu-0 eEWGUg"
            >
              <div
                class="sidebar___StyledDiv-sc-15m1zdu-1 flex items-center justify-start py-3 pr-4 transition-all pl-3"
              >
                <span
                  class="sidebar___StyledSpan-sc-15m1zdu-2 CYwdk"
                >
                  About
                </span>
              </div>
              <div
                class="collapse___StyledDiv-deqgwy-0 dISutg"
                role="root"
                style="min-height: 0px; max-height: 0px;"
              >
                <div
                  role="wrapper"
                />
              </div>
            </li>
          </ul>
        </aside>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
